packages/devtools_app/lib/src/shared/console/primitives/assignment.dart

```diff
@@ -0,0 +1,35 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Assignment of an item in console to a named variable.
+class ConsoleVariableAssignment {
+  ConsoleVariableAssignment._(this.consoleItemIndex, this.variableName);
+
+  /// Ziro based bottom up index for items in console.
+  final int consoleItemIndex;
+
+  final String variableName;
+
+  /// Parses expressions like `var x=$2`.
+  ///
+  /// Treats $_ as $0.
+  static ConsoleVariableAssignment? tryParse(String expression) {
+    const variableNameGroup = '([_a-zA-Z][_a-zA-Z0-9]{0,30})';
+    const indexGroup = '([_012345])';
```

Comment:
Do we only support indexes 0-5? I might have missed this in a previous PR, but is this documented somewhere?

Reply:
It is in compliance with chrome console: https://docs.google.com/document/d/1M65VISBzYwiySgjKU3iM_gvcDKUzOjxGCUlQWtHOtVM/edit#bookmark=id.u09mkakioulg

You normally just need 0 and 1.

Reply:
We should make sure this is documented somewhere in DevTools so it can be easily discovered by users.

Reply:
yes, this is the plan